Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-NewServiceFabricApplication.ps1

```diff
@@ -311,8 +311,8 @@
                     Write-Warning (Get-VstsLocString -Key SFSDK_CompressPackageWarning $InstalledSdkVersion)
                 }
             }
-            $global:operationId = $SF_Operations.CopyApplicationPackage
-            Copy-ServiceFabricApplicationPackage @copyParameters
+
+            Copy-ServiceFabricApplicationPackageAction -CopyParameters $copyParameters
             if (!$?)
```

Comment:
I am not sure whether it is necessary or not. Since it existed before my change, I am not going to remove this. May be it is there as Copy command throws a non-terminating error.

Reply:
Please go through this article https://blogs.technet.microsoft.com/heyscriptingguy/2011/05/12/powershell-error-handling-and-why-you-should-care/
We've used it because earlier we were directly calling SDK cmdlets (like Copy-ServiceFabricApplicationPackage) so there was some scope of call failures but now we're calling our own utility functions (like Copy-ServiceFabricApplicationPackageAction in this case) so there is no chance for call failure.
My suggestion is to remove this check from here and move it to Copy-ServiceFabricApplicationPackageAction function right after where you're calling Copy-ServiceFabricApplicationPackage cmdlet.

Do the same for all such checks!